src/components/OutOfOfficeForm.vue

```diff
@@ -253,11 +263,18 @@ export default {
 			} else {
 				this.enabled = state.enabled ? OOO_ENABLED : OOO_DISABLED
 			}
-			if (state.enabled) {
-				this.firstDay = state.start ? new Date(state.start) : new Date()
-				this.lastDay = state.end ? new Date(state.end) : null
+
+			if (state.enabled && state.start) {
+				this.firstDay = new Date(state.start)
+			}
+			if (state.enabled && state.end) {
+				this.lastDay = new Date(state.end)
+				// FIXME: The dav automation adds 23:59 and mail adds 24:00 hours to the last day.
+				//        Subtract 23 hours to get the actual date.
+				this.lastDay.setHours(this.lastDay.getHours() - 23, 0, 0, 0)
```

Comment:
Ugly but necessary at the moment. If the logic is fixed in dav and we don't support the old version any longer we can just subtract 24 hours.